tests/e2e/pkg/tests/mocks.go

```diff
@@ -0,0 +1,264 @@
+}
+
+// MockTestsProvider:
+// 3 different provider 3 different K8s cluster with 2 tests
+// Expected: 6 testCase 2 testCase on every provider
+// 3x2 + 3x3 = 15 test all-together
+// Runtime parallel: 3x(3) = 9sec (time of the longest testCase)
+func MockTestsProvider() Classifier {
+	k8sClusterPool := K8sClusterPool{}
+	k8sClusterPool.AddK8sClusters(
+		NewMockK8sCluster(
+			"testContextPath1",
+			"testContextName1",
+			"1.24",
+			"provider1",
+			"clusterID1",
+		),
+		NewMockK8sCluster(
+			"testContextPath2",
+			"testContextName2",
+			"1.24",
+			"provider2",
+			"clusterID2",
+		),
+		NewMockK8sCluster(
+			"testContextPath3",
+			"testContextName3",
+			"1.24",
+			"provider3",
+			"clusterID3",
+		),
+	)
+	return NewClassifier(k8sClusterPool, mockTest1, mockTest2)
+}
+
+// MockTestsProviderMoreTestsThenProvider:
+// 2 different provider 2 different K8s cluster with 3 tests
+// Expected: 6 testCase 3 testCase on every provider
+// 2x2 + 2x2 + 2x3 = 14 test all-together
+// Runtime parallel: 4 + 4 + 5 = 13
+func MockTestsProviderMoreTestsThenProvider() Classifier {
+	k8sClusterPool := K8sClusterPool{}
+	k8sClusterPool.AddK8sClusters(
+		NewMockK8sCluster(
+			"testContextPath1",
+			"testContextName1",
+			"1.24",
+			"provider1",
+			"clusterID1",
+		),
+		NewMockK8sCluster(
+			"testContextPath2",
+			"testContextName2",
+			"1.24",
+			"provider2",
+			"clusterID2",
+		),
+	)
+	return NewClassifier(k8sClusterPool, mockTest1, mockTest2, mockTest3)
+}
+
+// MockTestsVersionOne:
+// no different version 2 different K8s cluster with 2 tests
+// Expected: 2 testCase -> 1 testCase on each K8sCluster
+// 2x2  2x3 = 10 test all-together
+// Runtime parallel: 1x5 = 5
+func MockTestsVersionOne() Classifier {
+	k8sClusterPool := K8sClusterPool{}
+	k8sClusterPool.AddK8sClusters(
+		NewMockK8sCluster(
+			"testContextPath1",
+			"testContextName1",
+			"1.24",
+			"provider1",
+			"clusterID1",
+		),
+		NewMockK8sCluster(
+			"testContextPath2",
+			"testContextName2",
+			"1.24",
+			"provider2",
+			"clusterID2",
+		),
+	)
+	return NewClassifier(k8sClusterPool, mockTest1, mockTest2)
+}
+
+// MockTestsVersion:
+// 2 different version 3 different K8s cluster with 2 tests
+// Expected: 4 testCase -> 2 testCase on each version
+// 2x2  2x3 = 10 test all-together
+// Runtime parallel: 4 + 5 = 9
+func MockTestsVersion() Classifier {
+	k8sClusterPool := K8sClusterPool{}
+	k8sClusterPool.AddK8sClusters(
+		NewMockK8sCluster(
+			"testContextPath1",
+			"testContextName1",
+			"1.24",
+			"provider1",
+			"clusterID1",
+		),
+		NewMockK8sCluster(
+			"testContextPath2",
+			"testContextName2",
+			"1.24",
+			"provider2",
+			"clusterID2",
+		),
+		NewMockK8sCluster(
+			"testContextPath4",
+			"testContextName4",
+			"1.26",
+			"provider3",
+			"clusterID4",
+		),
+	)
+	return NewClassifier(k8sClusterPool, mockTest1, mockTest2)
+}
+
+// MockTestsVersion:
+// 2 different version 2 different version 4 K8s cluster with 2 tests
```

Comment:
Same as above, it'd be great if we can rephrase this

Reply:
fixed: [2b85e9f](https://github.com/banzaicloud/koperator/pull/1020/commits/2b85e9f4affdc52e1b2980983dcb990fa193bf2f)